File: client/src/lib/invoke/sendGpgKey.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { TokenType } from "@bindings";

export function sendGpgKey(
    gitlab_token_type: TokenType,
    gitlab_token: string,
    pub_key: string
): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("send_gpg_key", { gitlab_token_type, gitlab_token, pub_key }),
        (error) => `Failed to send GPG key: ${error}`
    );
}

File: client/src-tauri/src/commands/send_gpg_key.rs
use nexium::gitlab::{GitlabClient, TokenType};

#[tauri::command]
pub async fn send_gpg_key(
    gitlab_token_type: TokenType,
    gitlab_token: String,
    pub_key: String,
) -> Result<(), String> {
    tauri::async_runtime::spawn_blocking(move || {
        GitlabClient::new(gitlab_token, gitlab_token_type)
            .add_gpg_key(&pub_key)
            .map_err(|err| err.to_string())
    })
    .await
    .map_err(|err| format!("Failed to send GPG key: {}", err))?
}
